### PR TITLE
Fix corrupt UserTask icon when exporting to an image or printing

### LINKED ISSUE #515: 🐛 Printing Does Not Work Correctly Under Electron
#### Describe your issue

When printing the diagram with the Electron App the user task icons are not rendered correctly. The user task icons will look like this:

#### How can others reproduce the issue?

- Open up the BPMN-Studio Electron app
- Print any diagram with at least one user task
- Have a look at the user task icon

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by robpal94**
## What did you change?
Fixes #515 

This PR replaces the usage of the canvg libary to render and exporting a svg to a canvas since it seems to be broken (as referenced [here](https://github.com/canvg/canvg/issues/577) ).

## How can others test the changes?
* Create a Process with a UserTask
* Export this Process as an image or print it

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/modules/processdef-detail/processdef-detail.ts [hunk 1]
```diff
@@ -461,22 +494,54 @@ export class ProcessDefDetail {
...
+    imageElement.setAttribute('src', `data:image/svg+xml;base64, ${encodedSVG}`);
+
+    const returnPromise: Promise<string> = new Promise((resolve: any, reject: any): void => {
+      imageElement.onload = (): void => {
+        context.drawImage(imageElement, 0, 0, canvas.width, canvas.height);
+        const encodedImageURL: string = canvas.toDataURL(encoding);
+        resolve(encodedImageURL);
+      };
+
+      imageElement.onerror = (cause: any): void => {
```
Comment (SteffenKn) [inline src/modules/processdef-detail/processdef-detail.ts]: I think "cause" is not a good name for a variable.
Also please replace any.

File: src/modules/processdef-detail/processdef-detail.ts [hunk 2]
```diff
@@ -434,12 +462,17 @@ export class ProcessDefDetail {
    */
   public async _printDiagram(): Promise<void> {
     const svg: string = await this.bpmnio.getSVG();
-    const png: string = this._generateImageFromSVG('png', svg);
 
-    print.default({printable: png, type: 'image'});
+    try {
+      const png: string = await this._generateImageFromSVG('png', svg);
+      print.default({printable: png, type: 'image'});
+    } catch (error) {
```
Comment (SteffenKn) [inline src/modules/processdef-detail/processdef-detail.ts]: Add a type.

File: src/modules/processdef-detail/processdef-detail.ts [hunk 3]
```diff
@@ -434,12 +462,17 @@ export class ProcessDefDetail {
...
+      this._notificationService.showNotification(NotificationType.ERROR,
+        `An error occurred while processing the image for printing.`);
+    }
   }
 
-  private _generateImageFromSVG(desiredImageType: string, svg: any): string {
+  private async _generateImageFromSVG(desiredImageType: string, svg: any): Promise<string> {
```
Comment (SteffenKn) [inline src/modules/processdef-detail/processdef-detail.ts]: The correct type for svg is string.

File: src/modules/processdef-detail/processdef-detail.ts [hunk 4]
```diff
@@ -461,22 +494,54 @@ export class ProcessDefDetail {
...
+    canvas: HTMLCanvasElement,
+    context: CanvasRenderingContext2D,
+    encoding: string): Promise<string> {
+
+    const imageElement: HTMLImageElement = document.createElement('img');
+    const encodedSVG: string = btoa(svgContent);
+
+    imageElement.setAttribute('src', `data:image/svg+xml;base64, ${encodedSVG}`);
+
+    const returnPromise: Promise<string> = new Promise((resolve: any, reject: any): void => {
```
Comment (SteffenKn) [inline src/modules/processdef-detail/processdef-detail.ts]: We use Function as type for resolve and reject.

File: src/modules/processdef-detail/processdef-detail.ts (hunk 2 above)
Comment (Paulomart) [reply src/modules/processdef-detail/processdef-detail.ts]: Inside catch(...) the type is set by convention. It cannot be changed or set.

Comment (SteffenKn) [reply src/modules/processdef-detail/processdef-detail.ts]: That is wright my bad.

File: src/modules/processdef-detail/processdef-detail.ts [hunk 5]
```diff
@@ -461,22 +492,69 @@ export class ProcessDefDetail {
...
+    const loadImagePromise: Promise<string> = new Promise((resolve: Function, reject: Function): void => {
+      imageElement.onload = (): void => {
+        context.drawImage(imageElement, 0, 0, canvas.width, canvas.height);
+        const encodedImageURL: string = canvas.toDataURL(encoding);
+        resolve(encodedImageURL);
+      };
+
+      imageElement.onerror = (errorEvent: ErrorEvent): void => {
+        /*
+         * TODO: Find out if we can reject the promise with a more specify
```
Comment (Vyperus) [inline src/modules/processdef-detail/processdef-detail.ts]: specific

**Comment by robpal94** [comment]
I don't think, that this PR is useful anymore, since [this pr](https://github.com/process-engine/bpmn-studio/pull/523#pullrequestreview-134394237) introduce the dedicated Export and Printing Services, which already used the changes introduced by this PR. 

@alexanderkasten @Vyperus @cmg-dev

**Comment by Vyperus** [comment]
Closed in favor of #523.

**Pull request closed without merging**